pkg/kapp/clusterapply/converged_resource_factory.go

```diff
@@ -64,28 +64,29 @@ func (f ConvergedResourceFactory) New(res ctlres.Resource,
 		func(res ctlres.Resource, aRs []ctlres.Resource) (SpecificResource, []ctlres.ResourceRef) {
 			// Use newly provided associated resources as they may be modified by ConvergedResource
 			return ctlresm.NewAppsV1Deployment(res, aRs), []ctlres.ResourceRef{
-				{schema.GroupVersionResource{Group: "apps"}}, // for ReplicaSets
-				{schema.GroupVersionResource{Group: ""}},     // for Pods
+				{schema.GroupVersionResource{Group: "apps", Resource: "replicasets"}},
```

Comment:
Have we verified that these have their intended effect?
If we look at the code [here](https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/resources/refs.go#L18-L33), the logic that is used to match GVRs does not support matching a GVR with just Group and Resource values.

Reply:
Oops, it is added